pkg/router/metrics/haproxy/haproxy.go

```diff
@@ -593,7 +599,7 @@ func (e *Exporter) exportAndRecordRow(metrics metrics, rowID metricID, targetVal
 
 // exportCSVFields iterates over the returned CSV values and sets the appropriate metric in the map. Empty values or parse errors result in
 // no metric being scraped.
-func exportCSVFields(csvParseFailures prometheus.Counter, metrics metrics, baselineValues map[int]int64, updateBaseline bool, csvRow []string, labels []string) {
+func exportCSVFields(csvParseFailures prometheus.Counter, metrics metrics, baselineValues map[int]int64, counterMetrics []bool, csvRow []string, labels []string) {
```

Comment:
Replacing `updateBaseline` seems to change the documented (or at least implied) semantics of `Collect` versus `CollectNow`: https://github.com/openshift/router/blob/bf8bdff322db86a4536b1c5be4532a0ac5e06b0e/pkg/router/metrics/haproxy/haproxy.go#L140-L143
https://github.com/openshift/router/blob/bf8bdff322db86a4536b1c5be4532a0ac5e06b0e/pkg/router/metrics/haproxy/haproxy.go#L305-L308
https://github.com/openshift/router/blob/bf8bdff322db86a4536b1c5be4532a0ac5e06b0e/pkg/router/metrics/haproxy/haproxy.go#L332-L333

If my understanding is correct, the change would also change the documented semantics of `parseRow` and `exportAndRecordRow`: https://github.com/openshift/router/blob/bf8bdff322db86a4536b1c5be4532a0ac5e06b0e/pkg/router/metrics/haproxy/haproxy.go#L486-L490
https://github.com/openshift/router/blob/bf8bdff322db86a4536b1c5be4532a0ac5e06b0e/pkg/router/metrics/haproxy/haproxy.go#L577-L580

`CollectNow` calls `e.scrape(true)`, which calls `e.parseRow` with a non-nil value for `targetValues`, and `e.parseRow` then passes `targetValues` to `e.exportAndRecordRow`, which passes `targetValues` to `exportCSVFields`, which then updates `e.baseCounterValues[rowID]` (which is a map, so the update would mutate the state referenced by `e.baseCounterValues`).

In contrast, `Collect` calls `e.scrape(false)`, which calls `e.parseRow` with a nil value for `targetValues`, and `e.parseRow` passes the nil value to `e.exportAndRecordRow`, which passes the nil value to `exportCSVFields`, which then does **not** update `e.baseCounterValues`.  I think we need to keep `updateBaseline` to maintain the original semantics.

Reply:
Collect should not update base counter values.

Reply:
However, you're right that we shoud only update when not nil.  I restructured that to be clearer to intent.